### PR TITLE
v0 of razor html rendering + reqs fix & backend checks

### PR DESCRIPTION
**Pull request by iaincrawf**
Planning on using this to render html and send to pdf service for generating pdfs on backend. Can view the html in browser for debugging, see purchase orders module

- on reqs front end have swapped order of dept and nom as Les Baird spoke to me and thinks would be more clear
- Couple of missing checks to reqs backend: check part is not stock controlled (spoke to Andrew Mcv and they don't raise po reqs for stock controlled parts); check if ok to order - pl_orders pack had this but it just checks supplier is not closed or on hold, so checked these in C#

### CODE REVIEW

**Review by andrewfraser73 — APPROVED** [review]
LGTM

File: src/Service/Modules/PurchaseOrderModule.cs [hunk 1]
```diff
@@ -37,6 +38,8 @@ public void AddRoutes(IEndpointRouteBuilder app)
             app.MapGet("/purchasing/purchase-orders/tariffs", this.SearchTariffs);
             app.MapGet("/purchasing/purchase-orders", this.SearchPurchaseOrders);
             app.MapGet("/purchasing/purchase-orders/{orderNumber:int}", this.GetPurchaseOrder);
+            app.MapGet("/purchasing/purchase-orders/{orderNumber:int}/html", this.GetPurchaseOrderHtml);
+
```
Comment (RossStewart) [inline src/Service/Modules/PurchaseOrderModule.cs:42]: blank

File: tests/Integration/Integration.Tests/PurchaseOrderModuleTests/ContextBase.cs [hunk 2]
```diff
@@ -107,8 +85,8 @@ public void EstablishContext()
             this.PackagingGroupService = Substitute
                 .For<IFacadeResourceService<PackagingGroup, int, PackagingGroupResource, PackagingGroupResource>>();
             this.TariffService = Substitute.For<IFacadeResourceService<Tariff, int, TariffResource, TariffResource>>();
-            this.PurchaseOrderReqStateFacadeService =
-                Substitute.For<IFacadeResourceService<PurchaseOrderReqState, string, PurchaseOrderReqStateResource, PurchaseOrderReqStateResource>>();
+            this.PurchaseOrderReqStateFacadeService = Substitute
+                .For<IFacadeResourceService<PurchaseOrderReqState, string, PurchaseOrderReqStateResource, PurchaseOrderReqStateResource>>();
             this.MockReqDomainService = Substitute.For<IPurchaseOrderReqService>();
```
Comment (RossStewart) [inline tests/Integration/Integration.Tests/PurchaseOrderModuleTests/ContextBase.cs:91]: blank

**Review by RossStewart — APPROVED** [review]
Just 2 blanks but nothing looks sus to me!

**Pull request merged**